### PR TITLE
wayback-machine-archiver: init at 1.9.1

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/wayback-machine-archiver/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "agude";
+    repo = "wayback-machine-archiver";
+    rev = "v${version}";
+    sha256 = "0dnnqx507gpj8wsx6f2ivfmha969ydayiqsvxh23p9qcixw9257x";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [ pypandoc ];
+  propagatedBuildInputs = with python3.pkgs; [ requests ];
+  checkInputs = with python3.pkgs; [ pytest pytest-runner requests-mock ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/wayback-machine-archiver/default.nix:17]:
```suggestion
  checkInputs = with python3.pkgs; [ pytestCheckHook requests-mock ];
```

File: pkgs/tools/misc/wayback-machine-archiver/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ lib, python3, fetchFromGitHub }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "wayback-machine-archiver";
+  version = "1.9.1";
+
+  # pypi doesn't include tests
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/wayback-machine-archiver/default.nix:7]:
```suggestion
```

File: pkgs/tools/misc/wayback-machine-archiver/default.nix (hunk 1 above)
Comment (dali99) [reply pkgs/tools/misc/wayback-machine-archiver/default.nix:17]: 1. This doesn't work

2. Upstream tests by just running pytest -vv so I did that when the hook failed.

If you have any comments that apply to this PR and isn't just general linting advice, feel free to suggest more changes.

Comment (SuperSandro2000) [reply pkgs/tools/misc/wayback-machine-archiver/default.nix:17]: Then it should be patched to make work. pytest-runner does not work in the sandbox and has no use. Also pytestCheckHook does exactly what your checkPhase does.

If you paste some log I can provide more concrete help.

**Pull request merged**